File: swing_hello_worlds/window.clj
require "java"

(import
    '(java.awt EventQueue FlowLayout)
    '(java.awt.event ActionListener)
    '(javax.swing JButton JFrame JLabel JOptionPane)
)
  
(defn show-window []
    (let 
        [window (JFrame. "Example")
            label (JLabel. "Press: ")
            button (JButton. "Press me")
        ]
        (.addActionListener button
            (proxy [ActionListener] []
                (actionPerformed [evt]
                    (JOptionPane/showMessageDialog nil "I was pressed!")
                )
            )
        )
        (doto window
            (.setLayout (FlowLayout.))
            (.add label)
            (.add button)
            (.setSize 300 200)
            (.setDefaultCloseOperation JFrame/DISPOSE_ON_CLOSE)
            (.setVisible true)
        )
    )
)

(show-window)

File: swing_hello_worlds/window.js
var swing= JavaImporter();
swing.importPackage(Packages.javax.swing);
swing.importPackage(Packages.java.awt);
swing.importPackage(Packages.java.awt.event);
    
createFrame();

function createFrame() {
    var Frame = function() {};
    var frame;
    var label, button;
    var pressButton;

    Frame.prototype = new swing.JFrame("Example");
    frame = new Frame();

    frame.setLayout(new swing.FlowLayout());
    frame.add(label = new swing.JLabel("Press: "));
    frame.add(button = new swing.JButton("Press me"));

    button.addActionListener(function() {
        pressButton();
    });

    pressButton = function () {
        swing.JOptionPane.showMessageDialog(null, "I was pressed!");
    };

    frame.setSize(300, 200);
    frame.setDefaultCloseOperation(swing.JFrame.EXIT_ON_CLOSE);
    frame.setVisible(true);
    return frame;
};
